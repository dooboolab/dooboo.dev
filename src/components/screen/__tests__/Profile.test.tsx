import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

import Profile from '../Profile';
import { render, fireEvent, getByTestId } from '@testing-library/react';

const props = {
  history: {
    goBack: jest.fn(),
  },
};

describe('[Profile] render', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(<Profile />).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});

describe('[Profile] Interaction', () => {
  const component = <Profile {...props} />;
  let renderResult: any;

  beforeEach(() => {
    renderResult = render(component);
  });

  it('should simulate [onClick] when [btn] has been clicked', () => {
    const textInstance: any = renderResult.getByTestId('myText');
    expect(textInstance.textContent).toEqual('Profile');
  });
});
