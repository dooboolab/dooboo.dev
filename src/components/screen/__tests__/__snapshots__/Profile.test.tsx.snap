// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Profile] render renders without crashing 1`] = `
<div
  className="sc-bdVaJa jHgyWZ"
>
  <div
    data-testid="myText"
  >
    Profile
  </div>
</div>
`;
